tests/ToolInfoTest.php

```diff
@@ -21,13 +21,87 @@
  */
 final class ToolInfoTest extends \PHPUnit_Framework_TestCase
 {
+    const PHPUNIT_BIN = '/../vendor/bin/phpunit';
+
+    protected function setUp()
+    {
+        foreach ($this->provideScriptName() as $testArguments) {
+            $symlink = current($testArguments);
+            if ($symlink === self::PHPUNIT_BIN or is_link(__DIR__.$symlink)) {
+                continue;
+            }
+
+            symlink(__DIR__.self::PHPUNIT_BIN, __DIR__.$symlink);
+        }
+    }
+
+    protected function tearDown()
+    {
+        foreach ($this->provideScriptName() as $testArguments) {
+            $symlink = current($testArguments);
+            if ($symlink === self::PHPUNIT_BIN or !is_link(__DIR__.$symlink)) {
+                continue;
+            }
+
+            unlink(__DIR__.$symlink);
+        }
+    }
+
     public function testGetVersion()
     {
-        $this->assertInternalType('string', ToolInfo::getVersion());
+        $toolInfo = new ToolInfo(null);
+        $this->assertInternalType('string', $toolInfo->getVersion());
     }
 
     public function testIsInstallAsPhar()
     {
-        $this->assertFalse(ToolInfo::isInstalledAsPhar());
+        $toolInfo = new ToolInfo(null);
+        $this->assertFalse($toolInfo->isInstalledAsPhar());
+    }
+
+    /**
+     * @dataProvider provideScriptName
+     *
+     * @param string $scriptName
+     * @param mixed  $symlink
+     */
+    public function testIsInstalledByComposer($symlink)
+    {
+        if (!is_file(__DIR__.self::PHPUNIT_BIN)) {
+            $this->markTestSkipped(sprintf('Binary "%s" is necessary to emulate PHP-CS-Fixer installation through Composer', __DIR__.self::PHPUNIT_BIN));
+        }
+
+        $scriptName = __DIR__.$symlink;
+        if (!is_link($scriptName)) {
+            $this->markTestSkipped(sprintf('OS "%s" doesn\'t support the way the "%s" symlink is written', PHP_OS, $scriptName));
+        }
+
+        $toolInfo = new ToolInfo($scriptName);
+        $this->assertTrue($toolInfo->isInstalledByComposer());
+        $this->assertInternalType('string', $toolInfo->getVersion());
+
+        // Must be null because here we are in the project itself
+        // and can't be present in the composer installation
+        $this->assertNull($toolInfo->getComposerVersion());
+    }
+
+    public function provideScriptName()
+    {
+        return array(
+            array(self::PHPUNIT_BIN),
+            array('/Fixtures/symlinks/toolinfo-phpunit-symlink'),
+            array('/Fixtures/symlinks/toolinfo-phpunit-double-symlink'),
+        );
+    }
+
+    public function testGetComposerVersionThrowExceptionIfOutsideComposerScope()
+    {
+        $toolInfo = new ToolInfo('non_existent_script');
+
+        $this->assertFalse($toolInfo->isInstalledByComposer());
```

Comment:
`testIsInstalledAsPhar` is a bit misleading: currently it tests that it is NOT installed by Phar, because `'phar://' === substr(__DIR__, 0, 7)` is not mockable.

The test would look like:
```php
public function testIsInstallAsPhar()
{
    $toolInfo = new ToolInfo(null);
    $this->assertFalse($toolInfo->isInstalledAsPhar());
    $this->assertFalse($toolInfo->isInstalledByComposer());
}
```